db_test.go

```diff
@@ -996,6 +1035,30 @@ func TestWithPerconaDeadline(t *testing.T) {
 	}
 }
 
+func TestTransactionBlock(t *testing.T) {
+	db := makeTestDB(t, usersDDL)
+
+	defer db.Close()
+
+	if err := db.Transaction(func(tx *Tx) error {
+		_, err := tx.Exec("SELECT * from users")
```

Comment:
Can we test an UPDATE that errors (for example, there is a duplicate primary key)?

Reply:
Sure I'll add that test case.